lib/library.rb

```diff
@@ -0,0 +1,50 @@
+class Library
+  attr_reader :name, :books, :authors
+  def initialize(name)
+    @name = name
+    @books_array = []
+    @authors = []
+  end
+
+  def add_author(author)
+    @authors.push author
+  end
+
+  def books
+    @authors.each do |author|
+        author.books.each do |book|
+          book_being_added = book
+          @books_array.push book_being_added
+      # binding.pry
+      end
+    end
+    @books_array
+  end
+
+  def publication_time_frame_for(author)
```

Comment:
The logic for this method is good, but what might it look like if an author could tell you a list of their books sorted by publication year?